### DISCUSSION TITLE
How to deal with onSuccess or onError?

### DISCUSSION THREAD

**Discussion by NilsEvers** [DamianOsipiuk/vue-query#138] [Q&A] [answered]
useQuery is not providing any callbacks like `onSuccess` or `onError`, which is fine for simple use-cases where everything is handled in the template. But what if I want to call a function when an error has occurred, to create a notification for example.
Or how do I add the data to a store, when it was loaded successfully?

I could use `watch` to watch for changes. However, that would be quite cumbersome to do this everywhere.
Or I could  wrap everything in `watchEffect`...

How do you deal with use-cases like this? Is there a better way to archive this?

**Comment by wobsoriano** [comment, marked as answer]
How about doing something like this?

```ts
import { yourNotifFunc, doSomethingWithData } from './somewhere'

useQuery('todos', fetchTodoList, {
    onSuccess: (data) => {
        yourNotifFunc('success')
        doSomethingWithData(data)
    },
    onError: () => {
        yourNotifFunc('error')
    }
})
```

Also worth checking https://tkdodo.eu/blog/status-checks-in-react-query
